libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/session/MapboxTripSession.kt

```diff
@@ -82,17 +82,12 @@ internal class MapboxTripSession(
     ): NativeSetRouteResult {
         logD(
             "routes update (reason: ${setRoutesInfo.reason}, " +
-                "count: ${routes.size}) - starting",
+                "route IDs: ${routes.map { it.id }}) - starting",
             LOG_CATEGORY
         )
         isUpdatingRoute = true
         val result = when (setRoutesInfo) {
             is BasicSetRoutesInfo -> {
-                isOffRoute = false
-                invalidateLatestInstructions()
-                roadObjects = emptyList()
-                routeProgress = null
-                updateLegIndexJob?.cancel()
```

Comment:
what about the moment between successful route set and the first status update? Doe those fields keeps state of the previous route? Maybe we should also wait the next status update with the `origin=SET_ROUTE`?

Reply:
BTW, does NN specify the order of status update with `origin=SET_ROUTE` and callback from set route? @mapbox/navnative